CorsixTH/Lua/world.lua

```diff
@@ -971,16 +971,30 @@ end
 
 --! Dedicated function to allow unpausing by pressing 'p' again
 function World:pauseOrUnpause()
+  if self:systemPauseStatus() then return end -- System pause takes precedence
   if not self:isCurrentSpeed("Pause") then
     self:setSpeed("Pause")
   elseif self.prev_speed then
     self:setSpeed(self.prev_speed)
   end
 end
 
+--! Sets whether the game should be forcefully paused
+--!param state (bool)
+function World:systemPause(state)
+  self.system_pause = state
+end
+
+--! Reports the system pause status
+--!return (bool) true is system pause is active, else false
+function World:systemPauseStatus()
```

Comment:
This one isn't clear either, imho.

"system pause status" is a concept, a thing. A function is an action, something it changes, queries or constructs. Give a function a verb to express what it does.